### DISCUSSION TITLE
TCP/IP sockets via UART? (through AT-compatible transport)

### DISCUSSION THREAD

**Discussion by davidmcnabnz** [micropython/micropython#11120] [General]
Hi all,

Does anyone know of a library which can provide Python-style socket objects, back-ending out  through a UART to an AT commands compatible RS232 device?

Or do I need to write and PR it myself?

**Comment by rkompass** [comment]
I suppose that would be PPP, which we haven't generally available. If someone would be able to implement that, we could have networking over UART/USB which would be very nice.

**Reply by davidmcnabnz** [reply]
Ok I'll give it a shot. It would be nice to open up UART/AT-based networking on these little devices. Better yet if it can be done in a way that allows transport choice, and makes the transport transparent to code that's been using the WiFi modules.

**Reply by jokiefer** [reply]
any further working here?
Or did you found any other solution for this?

**Comment by rkompass** [comment]
Did you notice https://github.com/micropython/micropython/pull/14461 ?
Which is very good news!
Someone will have to compile that in with all the accompanying options/settings and test its usability.

**Reply by jokiefer** [reply]
Thanks. 
Am i right that this is only the registration possibility to register uart as ip based socket?
The AT command handling is not part of it right? That have to implemented elsewhere right?
